core/src/main/java/org/apache/iceberg/deletes/BitmapPositionDeleteIndex.java

```diff
@@ -92,4 +107,113 @@ public Collection<DeleteFile> deleteFiles() {
   public long cardinality() {
     return bitmap.cardinality();
   }
+
+  /**
+   * Serializes the index using the following format:
+   *
+   * <ul>
+   *   <li>The length of the magic bytes and bitmap stored as 4 bytes (big-endian).
+   *   <li>A 4-byte {@link #MAGIC_NUMBER} (little-endian).
+   *   <li>The bitmap serialized using the portable Roaring spec (little-endian).
+   *   <li>A CRC-32 checksum of the magic bytes and bitmap as 4-bytes (big-endian).
+   * </ul>
+   *
+   * Note that the length and the checksum are computed for the bitmap data, which includes the
+   * magic bytes and bitmap for compatibility with Delta.
+   */
+  @Override
+  public ByteBuffer serialize() {
```

Comment:
Puffin writers currently assume we will create and pass `ByteBuffer`. We also compute a checksum, so I am allocating a byte array upfront.